javacutil/src/main/java/org/checkerframework/javacutil/AnnotationBuilder.java

```diff
@@ -86,8 +86,18 @@ public AnnotationBuilder(ProcessingEnvironment env, CharSequence name) {
     }
 
     public AnnotationBuilder(ProcessingEnvironment env, AnnotationMirror annotation) {
-        this.elements = env.getElementUtils();
-        this.types = env.getTypeUtils();
+        this(env.getElementUtils(), env.getTypeUtils(), annotation);
+    }
+
+    // TODO: This is a bit gross, is used by UnitsRelationsTools.
+    /** Leaves type field null, should not be used if that field is needed. */
+    public AnnotationBuilder(Elements elements, AnnotationMirror annotation) {
```

Comment:
I don't find this new constructor appropriate. There is no indication how to choose between this constructor and the one just before, which takes a `ProcessingEnvironment` instead of a `Elements`.
It sets a field to null, although the field should be non-null. So this creates an `AnnotationBuilder` that might result in NPEs when some operations are performed on it.

Reply:
This is used only by UnitsRelationsTools (see comment on line 92).  I didn't see how to obtain a `ProcessingEnvironment` in UnitsRelationsTools.  If you can suggest a way to do that, then we can remove this constructor -- like you, I would prefer that.  Otherwise, this is needed by UnitsRelationsTools.